contracts/libraries/Math.sol

```diff
@@ -80,6 +97,288 @@ library Math {
+            log_mX128 = (log_mX128 * x3_2) / 2 ** 128 + 983502807631630754645846664098547106; //int256(512 << 128) / int256(177147);
+            log_mX128 = (log_mX128 * x3_2) / 2 ** 128 - 1659660987878376898464866245666298242; //int256(32 << 128) / int256(6561);
+            log_mX128 = (log_mX128 * x3_2) / 2 ** 128 + 2845133122077217540225484992570796986; //int256(128 << 128) / int256(15309);
+            log_mX128 = (log_mX128 * x3_2) / 2 ** 128 - 4978982963635130695394598736998894726; //int256(32 << 128) / int256(2187);
+            log_mX128 = (log_mX128 * x3_2) / 2 ** 128 + 8962169334543235251710277726598010507; //int256(32 << 128) / int256(1215);
+            log_mX128 = (log_mX128 * x3_2) / 2 ** 128 - 16804067502268566096956770737371269701; //int256(4 << 128) / int256(81);
+            log_mX128 = (log_mX128 * x3_2) / 2 ** 128 + 33608135004537132193913541474742539403; //int256(8 << 128) / int256(81);
+            log_mX128 = (log_mX128 * x3_2) / 2 ** 128 - 75618303760208547436305468318170713656; //int256(2 << 128) / int256(9);
+            log_mX128 = (log_mX128 * x3_2) / 2 ** 128 + 226854911280625642308916404954512140970; //int256(2 << 128) / int256(3);
+            log_mX128 =
+                (log_mX128 * x3_2) /
+                2 ** 128 +
+                int256(137972626690900373465550041896316339718); // Log[3/2]*2**128
+
+            // compute expression for  log[m] + (msg-96)*Log[2] * 2**128
+            int256 numerator = log_mX128 +
+                int256(msb - 96) *
+                int256(235865763225513294137944142764154484399);
+
+            int256 tick;
+            int256 tickDown;
+            // round up if tick is negative
+            if (msb < 96) {
+                tick =
+                    (numerator +
+                        8506633848419547728916991634454851 +
+                        1701326769683909545783398326890970) /
+                    17013267696839095457833983268909703 -
+                    1;
+                tickDown =
+                    (numerator +
+                        8506633848419547728916991634454851 -
+                        1701326769683909545783398326890970) /
+                    17013267696839095457833983268909703;
+            } else {
+                tick =
+                    (numerator + 1701326769683909545783398326890970) /
+                    17013267696839095457833983268909703;
+                tickDown =
+                    (numerator - 1701326769683909545783398326890970) /
+                    17013267696839095457833983268909703;
+            }
+
+            if (tick == tickDown) {
+                return int24(tick);
+            } else {
+                if (sqrtPriceX96 < getSqrtRatioAtTick(int24(tick))) {
+                    tick -= 1;
+                }
+                return int24(tick);
+            }
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            LIQUIDITY AMOUNTS (STRIKE+WIDTH)
```

Comment:
fix formatting, i recommend https://github.com/transmissions11/headers for centering these types of headers :)